explainer.md

```diff
@@ -1,624 +1,397 @@
-*   When the callback returns, the new DOM is considered ready for first render.
-    This starts step 5 onwards to create new pseudo elements and start
-    animations.
+Element selectors:
 
-## Additional Script APIs
+- `::page-transition-container(name)` - Select the transition part containers of a given `page-transition-tag`.
+- `::page-transition-part(name)` - Select the transition parts of a given `page-transition-tag`.
+- `::page-transition-texture-outgoing(name)` - Select the outgoing texture of a given `page-transition-tag`.
+- `::page-transition-texture-incoming(name)` - Select the incoming texture of a given `page-transition-tag`.
+- `::page-transition-root-outgoing` - Select the outgoing root texture.
+- `::page-transition-root-incoming` - Select the incoming root texture.
+- `::page-transition-background` - This can be set for cases where something needs to be rendered underneath the root textures.
 
-The following example shows how developers can configure the transition in
-script for an MPA.
+These will be selecting a pseudo-element, or an element in a UA-created shadow DOM.
 
-### Old Document
+```css
+::page-transition-container(header) {
+  animation-delay: 300ms;
+}
+```
+
+CSS can be used to make changes to the automatic animation, or override `animation-name` to remove the default.
+
+### Web animation API
 
 ```js
-addEventListener("navigate", (event) => {
-  // Add sharedid attribute to elements to offer for the transition
-  // based on current document state.
-  document.querySelector(".header").sharedid="header-id";
-
-  // setData can be used to pass opaque contextual information to the
-  // new page. The argument type is |any|.
-  document.documentTransition.setData({ version: 123 });
+// In Page-B
+document.performTransition((transition) => {
+  // Build up transition parts, then…
+  transition.root.querySelector('[part=header]').animate(…);
 });
 ```
 
-### New Document
+- Open question: What's the deadline for calling `performTransition`?
+
+Again, this is a half-baked sketch. This example assumes that elements in the shadow root are given a part attribute, but there's likely some better way to address the elements.
+
+If the 'stage' of the transition is exposed as a shadow root, the developer can interact with the elements in a regular way. The developer could even create elements specifically for the transition.
+
+## Signaling the end of a page transition
+
+At the start of the transition, the browser could gather all the `Animation`s active on the stage, and assume the animation is complete once all animations finish.
+
+In addition, the JS API could include a way to provide a promise which keeps the transition active, allowing for animations driven some other way, such as `requestAnimationFrame`.
+
+## Example
+
+Using the sketches above, here's how the example Page-A to Page-B transition could be done:
+
+https://user-images.githubusercontent.com/93594/141100217-ba1fa157-cd79-4a9d-b3b4-67484d3c7dbf.mp4
```

Comment:
Can you also add an example for what this would look like if the capture was in single image mode?

Reply:
Ohh, shall do. I think it'd make more sense earlier in the document, just before the section on the other mode.